src/mqt/bench/tket_helper.py

```diff
@@ -310,16 +311,21 @@ def get_mapped_level(
     native_gate_set_rebase = get_rebase(gate_set_name)
     arch = Architecture(cmap)
 
+    # add blank wires to the circuit such that afterwards the number of qubits is equal to the number of qubits of the architecture
+    highest_used_qubit_index = max(max(sublist) for sublist in cmap)
+    diff = highest_used_qubit_index + 1 - qc_tket.n_qubits  # offset of one is added because the indices start at 0
+    qc_tket.add_blank_wires(diff)
+
     native_gate_set_rebase.apply(qc_tket)
     FullPeepholeOptimise(target_2qb_gate=OpType.TK2).apply(qc_tket)
     placer = LinePlacement(arch) if lineplacement else GraphPlacement(arch)
-    PlacementPass(placer).apply(qc_tket)
-    RoutingPass(arch).apply(qc_tket)
-    FullPeepholeOptimise(target_2qb_gate=OpType.TK2, allow_swaps=False).apply(qc_tket)
-    native_gate_set_rebase.apply(qc_tket)
-    if not qc_tket.valid_connectivity(arch, directed=True):
-        CXMappingPass(arc=arch, placer=placer, directed_cx=True, delay_measures=False).apply(qc_tket)
+    CXMappingPass(arc=arch, placer=placer, directed_cx=True, delay_measures=False).apply(qc_tket)
```

Comment:
What's the reason for this pass to now be applied unconditionally? Shouldn't this just be applied when the connectivity is not satisfied after the standard pass?

Asked differently: shouldn't this special treatment only apply for directed architectures?

Reply:
I thought always using the `CXMappingPass` would remove the `if` condition and, by that, simplify the code. Do you see a disadvantage in using that MappingPass as the default one?

Reply:
Yeah. I would guess that the default mapping pass is superior to the `CXMappingPass` otherwise the `CXMappingPass` would be used per default in TKET.

Reply:
Hm, I see. To be honest I thought the `CXMappingPass` would be more powerful (maybe I was misled because `CXMappingPass` sounded more sophisticated to me than `RoutingPass`). I adapted the code accordingly.